src/xtgeo/grid3d/_gridprops_import_eclrun.py

```diff
@@ -20,6 +23,45 @@
 
 logger = null_logger(__name__)
 
+if TYPE_CHECKING:
+    from xtgeo.common.sys import _XTGeoFile
+
+
+def read_eclrun_properties(xtg_file: _XTGeoFile) -> Generator[str, None, None]:
+    """Generates property names from Eclipse files.
+
+    Args:
+        xtg_file: The _XTGeoFile representing an Eclipse INIT or restart file.
+
+    Returns:
+        Names of properties within the Eclipse file.
+    """
+    is_stream = isinstance(xtg_file.file, (io.BytesIO, io.StringIO))
+    try:
+        if is_stream:
+            mark = xtg_file.file.tell()
+        ntotal = 0
+        nactive = 0
+        for item in resfo.lazy_read(xtg_file.file):
+            keyword = item.read_keyword().strip()
+
+            if keyword == "INTEHEAD":
+                data = item.read_array()
+                # nx * ny * nz
+                ntotal = data[8] * data[9] * data[10]
+                nactive = data[11]
+                continue
+
+            # Do some simple filtering of things that cannot be properties
+            if (
+                item.read_length() in (ntotal, nactive)
+                and item.read_type() in (b"INTE", b"REAL", b"DOUB")
+            ):
+                yield item.read_keyword().strip()
```

Comment:
Maybe just return `keyword` 
```suggestion
                yield keyword
```